### PR TITLE
Add refactored code

### PR DESCRIPTION
**Pull request by NatiDeme**
# Changes in this PR:
- No linter errors.
- Followed SOLID principle
- Followed YAGNI, KISS, and DRY rules
- Logic has been separated

### CODE REVIEW

File: create_item.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    case gets.chomp.downcase
+    when 'y'
+      parent_permission = true
+
+    when 'n'
+      parent_permission = false
+    else
+      puts('please enter [Y/N]: ')
+    end
```
Comment (thecodechaser) [inline create_item.rb:37]: - [x] In the `create_student` method you are getting input from users and creating the student, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.

File: create_item.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    new_teacher = Teacher.new(specialization, age, name)
+    @people << new_teacher
+    puts('Teacher person created successfully!')
+    run
+  end
```
Comment (thecodechaser) [inline create_item.rb]: - [x] In the `create_teacher` method you are getting input from users and creating the teacher, This method does not follow the `single responsibility principle. Please make sure to separate these operations into different methods to achieve the `single responsibility principle.

File: create_item.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    puts 'Author:'
+    author = gets.chomp
+
+    new_book = Book.new(title, author)
+    @book << new_book
+
+    puts('Book created successfully')
+    run
+  end
```
Comment (thecodechaser) [inline create_item.rb]: - [x] In the `create_book` method you are getting input from users and creating the book, This method does not follow the `single responsibility principle`. Please make sure to separate these operations into different methods to achieve the `single responsibility principle`.

File: create_item.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    puts selected_person
+
+    print('Date: ')
+    rental_date = gets.chomp
+
+    new_rental = Rental.new(rental_date, selected_book, selected_person)
+    @rentals << new_rental
+    run
+  end
+end
```
Comment (thecodechaser) [inline create_item.rb]: - [x] In the `create_rental` method you are getting input from users and creating the rental, This method does not follow the `single responsibility principle`. Please make sure to separate these operations into different methods to achieve the `single responsibility principle`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @yohannesdagnachew, and @NatiDeme👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @NatiDeme @yohannesdagnachew,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**